kedro/io/data_catalog.py

```diff
@@ -42,10 +43,16 @@ def _get_credentials(
     Raises:
         KeyError: When a data set with the given name has not yet been
             registered.
+        MissingConfigError: When no credentials file is found in the
+            project.
 
     """
     try:
         return credentials[credentials_name]
+    except TypeError as exc:
+        raise MissingConfigException(
+            f"Credentials not found in your Kedro project config."
+        )
```

Comment:
Would it be better if this moved before the `try` block? When this function is called but `credentials` is None then it indicates it doesn't have the configuration.